backend/src/main/java/com/google/sps/servlets/FindRecipesServlet.java

```diff
@@ -103,4 +120,23 @@ private boolean isDietFriendly(Recipe recipe, List<String> diets, List<String> a
     }
     return true;
   }
+
+  private List<Recipe> getRandomisedRecipes(List<Recipe> recipes) {
+    if (recipes.size() <= MAX_NUMBER_OF_RECIPES_TO_RETURN) {
+      return recipes;
+    }
+
+    List<Recipe> result = new ArrayList<>();
+    Random random = new Random();
+
+    for (int i = 0; i < MAX_NUMBER_OF_RECIPES_TO_RETURN; i++) {
+      if (recipes.isEmpty()) {
+        return result;
+      }
+      int index = random.nextInt(recipes.size());
+      result.add(recipes.get(index));
+      recipes.remove(index);
```

Comment:
It's bad practice to modify the object passed in by the caller, because the caller might not be expecting it.
You can use Collections.shuffle(recipes); instead and take the first m numbers you need.